spanner/transaction.go

```diff
@@ -952,6 +952,28 @@ func (t *ReadWriteTransaction) begin(ctx context.Context) error {
 	return err
 }
 
+// Commit tries to commit a readwrite transaction to Cloud Spanner. It also
+// returns the commit timestamp for the transactions.
+//
+// This method should be used with client.BeginReadWriteTransaction and only be
+// used if a custom error handling is needed. For normal use cases,
+// client.ReadWriteTransaction should be used because it has robust error
+// handlings and retries.
+func (t *ReadWriteTransaction) Commit(ctx context.Context) (time.Time, error) {
```

Comment:
This method should not be defined on `ReadWriteTransaction`. Instead, we should define a separate type that is used for these statement based transactions. Otherwise, this method will be visible to anyone using the normal `Client.ReadWriteTransaction` method. And even though the documentation states that a user should not use this method when using the normal `Client.ReadWriteTransaction` method, there will probably be many who do not see / understand this, and still try to use it, causing unpredictable errors.